docs/guides/secrets.md

```diff
@@ -0,0 +1,177 @@
+    with snow_connector as connector:
+        connector.execute(
+            "CREATE TABLE IF NOT EXISTS customers (name varchar, address varchar);"
+        )
+        connector.execute_many(
+            "INSERT INTO customers (name, address) VALUES (%(name)s, %(address)s);",
+            seq_of_parameters=[
+                {"name": "Ford", "address": "Highway 42"},
+                {"name": "Unknown", "address": "Space"},
+                {"name": "Me", "address": "Myway 88"},
+            ],
+        )
+
+
+@task
+def fetch_data(snow_connector: SnowflakeConnector) -> list:
+    all_rows = []
+    with snow_connector as connector:
+        while True:
+            new_rows = connector.fetch_many("SELECT * FROM customers", size=2)
+            if len(new_rows) == 0:
+                break
+            all_rows.append(new_rows)
+    return all_rows
+
+
+@flow(log_prints=True)
+def snowflake_flow():
+    aws_secret_block = AwsSecret.load("my-snowflake-pw")
+
+    snow_connector = SnowflakeConnector(
+        schema="MY_SCHEMA",
+        database="MY_DATABASE",
+        warehouse="COMPUTE_WH",
+        fetch_size=1,
+        credentials=SnowflakeCredentials(
+            role="MYROLE",
+            user="MYUSERNAME",
+            account="ab12345.us-east-2.aws",
+            password=json.loads(aws_secret_block.read_secret()).get("my-snowflake-pw"),
+        ),
+        poll_frequency_s=1,
+    )
+
+    setup_table(snow_connector)
+    all_rows = fetch_data(snow_connector)
+    print(all_rows)
+
+
+if __name__ == "__main__":
+    snowflake_flow()
+```
+
+Fill in the relevant details for your Snowflake account and run the script.
+
+Note that the flow reads the Snowflake password from the AWS Secret Manager and uses it in the `SnowflakeCredentials` block.
+The `SnowflakeConnector` block uses the nested `SnowflakeCredentials` block to connect to Snowflake.
+Again, neither of the Snowflake blocks are saved, so the credentials are not stored in Prefect Cloud.
+
+Check out the [`prefect-snowflake` docs](/integrations/prefect-snowflake) for more examples of working with Snowflake.
+
+## Next steps
+
+Now you can turn your flow into a [deployment](/guides/prefect-deploy/) so that you and your team can run it remotely on a schedule, in response to an event, or manually.  
+
+Make sure to specify the `prefect-aws` and `prefect-snowflake` dependencies in your work pool or deployment so that they are available to runtime.
+
+Also ensure your compute has the AWS credentials for accessing the secret in AWS Secrets Manager.
+
+You've seen how to use Prefect blocks to store non-sensitive configuration and fetch sensitive configuration values from the environment.
+You can use this pattern to connect to other third-party services that require credentials, such as databases and APIs.
+You can use a similar pattern with any cloud provider secret manager, or extend it to work with environment variables.
```

Comment:
```suggestion
You can use a similar pattern with any secret manager, or extend it to work with environment variables.
```

Reply:
I wouldn't describe Vault as a cloud provider, and I think its one of the most common secrets managers